### ISSUE TITLE
"All types could not be fully resolved"

### ISSUE THREAD

**Issue by malmaud** [JuliaIO/ProtoBuf.jl#71]
I'm getting this error when trying to compile the tensorflow protobufs. Not sure where to start debugging:

```
~/Dropbox/code/arrayflow/src> protoc -I=/Users/malmaud/tensorflow --julia_out=. /Users/malmaud/tensorflow/tensorflow/core/framework/graph.proto
INFO: Recompiling stale cache file /Users/malmaud/.julia/lib/v0.4/ProtoBuf.ji for module ProtoBuf.
WARNING: Base.String is deprecated, use AbstractString instead.
  likely near no file:0
WARNING: Base.String is deprecated, use AbstractString instead.
  likely near no file:0
--julia_out: All types could not be fully resolved:
tensorflow.NameAttrList_AttrEntry depends on tensorflow.AttrValue
tensorflow.NameAttrList depends on tensorflow.NameAttrList_AttrEntry
tensorflow.AttrValue depends on tensorflow.NameAttrList
```

**Comment by tanmaykm** [comment]
Unable to look at the proto file now, but this is probably #61.

One way to work around it is to tweak the spec or the `meta` attributes a bit.
I had to do something similar here: https://github.com/tanmaykm/ProtoBuf.jl/blob/master/gen/google/protobuf/struct.proto#L43-L62

**Comment by malmaud** [comment]
Oh you're right, there is a recursive relationship between `AttrValue` and `NameAttrList` messages:


```
message AttrValue {
  ...
    NameAttrList func = 10;
...
}

// A list of attr names and their values. The whole list is attached
// with a string name.  E.g., MatMul[T=float].
message NameAttrList {
  string name = 1;
  map<string, AttrValue> attr = 2;
}
```

**Comment by tanmaykm** [comment]
fixed by #100
